Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -119,4 +124,52 @@ async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingProfile,
     }
 }
 
+/**
+ * Executes Web Deploy command
+ * 
+ * @param   webDeployPkg                   Web deploy package
+ * @param   webAppName                      web App Name
+ * @param   publishingProfile               Azure RM Connection Details
+ * @param   removeAdditionalFilesFlag       Flag to set DoNotDeleteRule rule
+ * @param   excludeFilesFromAppDataFlag     Flag to prevent App Data from publishing
+ * @param   takeAppOfflineFlag              Flag to enable AppOffline rule
+ * @param   virtualApplication              Virtual Application Name
+ * @param   setParametersFile               Set Parameter File path
+ * @param   additionalArguments             Arguments provided by user
+ * 
+ */
+async function DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishingProfile, virtualApplication, isFolderBasedDeployment) {
+
+    var isDeploymentSuccess = true;
+    var deploymentError = null;
+    try {
+        
+        var virtualApplicationMappings = azureWebAppDetails.properties.virtualApplications;
+        var webAppZipFile = webDeployPkg;
+        if(isFolderBasedDeployment){
+            webAppZipFile = tl.getVariable('System.DefaultWorkingDirectory') + randomstring.generate(7) + '.zip';
```

Comment:
We could have some hardcoded name for new zip package (like 'temp_web_pack.zip') instead of generating one .